### PR TITLE
Added security_policy_id attribute to twingate_group

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/257

## Changes
- added new optional attribute `security_policy_id`

### CODE REVIEW

File: twingate/internal/provider/resource/group.go [hunk 1]
```diff
@@ -26,6 +26,12 @@ func Group() *schema.Resource {
 				Description: "The name of the group",
 			},
 			// computed
+			"security_policy_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Optional:    true,
+				Description: "Defines which security policy applies to this group. The security policy ID can be obtained from the twingate_security_policy and twingate_security_policies data sources.",
```
Comment (vmanilo) [inline twingate/internal/provider/resource/group.go:33]: @alexmensch please check doc string

File: docs/resources/group.md [hunk 2]
```diff
@@ -30,6 +30,10 @@ resource "twingate_group" "aws" {
 
 - `name` (String) The name of the group
 
+### Optional
+
+- `security_policy_id` (String) Defines which security policy applies to this group. The security policy ID can be obtained from the twingate_security_policy and twingate_security_policies data sources.
```
Comment (alexmensch) [inline docs/resources/group.md:35]: Looks good, just need to capitalize our product terms.

--> "Defines which Security Policy applies to this Group. The Security Policy ID can be obtained from the `twingate_security_policy` and `twingate_security_policies` data sources."

**Review by alexmensch — CHANGES_REQUESTED** [review]

File: docs/resources/group.md (hunk 2 above)
Comment (vmanilo) [reply docs/resources/group.md]: updated

File: twingate/internal/provider/resource/resource.go [hunk 3]
```diff
@@ -523,15 +523,27 @@ func convertResource(data *schema.ResourceData) (*model.Resource, error) {
...
 		RemoteNetworkID: data.Get(attr.RemoteNetworkID).(string),
 		Address:         data.Get(attr.Address).(string),
 		Protocols:       protocols,
 		Groups:          convertGroups(data),
 		ServiceAccounts: convertServiceAccounts(data),
 		IsAuthoritative: convertAuthoritativeFlag(data),
-	}, nil
+	}
+
+	isVisible, ok := data.GetOkExists(attr.IsVisible) //nolint
```
Comment (ekampf) [inline twingate/internal/provider/resource/resource.go:536]: Why are these changes part of this PR?

File: twingate/internal/test/acctests/resource/resource_test.go [hunk 4]
```diff
@@ -1539,8 +1539,7 @@ func TestAccTwingateCreateResourceWithFlagIsVisible(t *testing.T) {
 				),
 			},
 			{
-				ExpectNonEmptyPlan: true,
-				Config:             createResourceWithFlagIsVisible(terraformResourceName, remoteNetworkName, resourceName, false),
+				Config: createResourceWithFlagIsVisible(terraformResourceName, remoteNetworkName, resourceName, false),
```
Comment (ekampf) [inline twingate/internal/test/acctests/resource/resource_test.go:1542]: Why are these changes part of this PR?

**Review by ekampf — CHANGES_REQUESTED** [review]
Check regarding the `resource` changes

File: twingate/internal/provider/resource/resource.go (hunk 3 above)
Comment (vmanilo) [reply twingate/internal/provider/resource/resource.go:536]: extracted this changes into separate PR

File: twingate/internal/test/acctests/resource/resource_test.go (hunk 4 above)
Comment (vmanilo) [reply twingate/internal/test/acctests/resource/resource_test.go]: moved this changes to separate PR

**Review by ekampf — APPROVED** [review]

**Pull request merged**